### PR TITLE
[EXPERIMENT] add support for checking types with Flow

### PR DESCRIPTION
**Pull request by shiftkey**
I'm not sure if this is appealing to you, but as part of my investigation into :fire:ing the PortableGit dependency I started down the path of what it would take to get flow incorporated (so I don't break the callers in the app).

TL;DR:
 - I now have `flow` testing the `spawnGit` method and where it's being used
 - I made the method signature more explicit rather than get too deep into flow to handle the magic of JS. I plan to come back to that.
 - I added `electron-compile` to handle the transpiling which needs to remove the flow type information, so that we don't need to worry about changing the packaging steps
 - `standard` now breaks when testing these annotations - I could swap this out to use the Node API and pass in the transpiled output, but that would mean a standalone script rather than a simple command

Ship List:
 - [x] annotate `spawnGit` and it's callers to verify the type information
 - [x] update app so that it strips type information at runtime
 - [x] run tests for `flow` and `standard` as part of `npm test`
 - [x] cleanup existing usages of `spawnGit` to drop unnecessary `execOpts` parameters

### CODE REVIEW

**Comment by shiftkey** [comment]
Oh, and using flow was how I found #175 😁

File: lib/spawn-git.js [hunk 1]
```diff
@@ -8,16 +10,33 @@ var exec = require('child_process').exec
...
-var winGit = path.join(__dirname, '../assets/PortableGit/bin/git.exe  ')
+var winGit = path.join(__dirname, '../assets/PortableGit/bin/git.exe')
+
+type ErrorHandler = (error: ?Error, stdout: string, stderr: string) => void
 
-module.exports = function spawnGit (command, options, callback) {
-  if (typeof options === 'function') {
-    callback = options
-    options = null
+function getText(stdout: Buffer | string): string {
```
Comment (shiftkey) [inline lib/spawn-git.js]: This code feels really :shit: but I haven't really dug into whether it's correct here to be paranoid about `exec` returning a `Buffer`

**Comment by shiftkey** [comment]
Let me know if you'd like the Travis changes extracted to a different PR

**Comment by jlord** [comment]
Oh dang, I hate that you did all this work! In general I'm anti things that you have to precompile so I wouldn't want to to add Flow or Babel. I'd rather have tests help find errors like the one with `request`. There are no tests in this project 😱 I'd like to have simple tests with tape/tap to tests the verify scripts but that always gets pushed to the bottom of my list.

I am, however, **very** interested in :fire: or reducing Portable Git, though I'm not sure how Flow/Babel would help with that? Can you explain more about how you were thinking it could be 🔥 'd?

**Comment by shiftkey** [comment]
@jlord don't sweat it - it was a fun learning experience for me.

> There are no tests in this project 😱 I'd like to have simple tests with tape/tap to tests the verify scripts but that always gets pushed to the bottom of my list.

That's fine. I'll have a look into where I could help out with that.

> I am, however, very interested in 🔥 or reducing Portable Git, though I'm not sure how Flow/Babel would help with that? Can you explain more about how you were thinking it could be 🔥 'd?

The project itself uses Typescript and removes the dependency of having Git on your PATH - it also supports Windows/macOS/Ubuntu, so it feels like a great fit for Git-It. I wanted to use this exercise as a way to catch regressions as I went through the refactoring exercise, but given how specific `spawnGit` is I think I can live without needing Flow/Babel.

**Pull request closed without merging**